instrumentation/host/host.go

```diff
@@ -0,0 +1,279 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host // import "go.opentelemetry.io/contrib/instrumentation/host"
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/mem"
+	"github.com/shirou/gopsutil/net"
+	"github.com/shirou/gopsutil/process"
+
+	"go.opentelemetry.io/contrib"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/api/unit"
+)
+
+// Host reports the work-in-progress conventional host metrics specified by OpenTelemetry
+type host struct {
+	config Config
+	meter  metric.Meter
+}
+
+// Config contains optional settings for reporting host metrics.
+type Config struct {
+	// MeterProvider sets the metric.Provider.  If nil, the global
+	// Provider will be used.
+	MeterProvider metric.Provider
+}
+
+// Option supports configuring optional settings for host metrics.
+type Option interface {
+	// ApplyHost updates *Config.
+	ApplyHost(*Config)
+}
+
+// WithMeterProvider sets the Metric implementation to use for
+// reporting.  If this option is not used, the global metric.Provider
+// will be used.  `provider` must be non-nil.
+func WithMeterProvider(provider metric.Provider) Option {
+	return metricProviderOption{provider}
+}
+
+type metricProviderOption struct{ metric.Provider }
+
+// ApplyHost implements Option.
+func (o metricProviderOption) ApplyHost(c *Config) {
+	c.MeterProvider = o.Provider
+}
+
+var (
+	// Label sets for CPU time measurements.
+
+	LabelCPUTimeUser   = []kv.KeyValue{kv.String("state", "user")}
+	LabelCPUTimeSystem = []kv.KeyValue{kv.String("state", "system")}
+	LabelCPUTimeOther  = []kv.KeyValue{kv.String("state", "other")}
+	LabelCPUTimeIdle   = []kv.KeyValue{kv.String("state", "idle")}
+
+	// Label sets used for Memory measurements.
+
+	LabelMemoryAvailable = []kv.KeyValue{kv.String("state", "available")}
+	LabelMemoryUsed      = []kv.KeyValue{kv.String("state", "used")}
+
+	// Label sets used for Network measurements.
+
+	LabelNetworkTransmit = []kv.KeyValue{kv.String("direction", "transmit")}
+	LabelNetworkReceive  = []kv.KeyValue{kv.String("direction", "receive")}
+)
+
+// Configure computes a Config from the supplied Options.
+func Configure(opts ...Option) Config {
+	c := Config{
+		MeterProvider: global.MeterProvider(),
+	}
+	for _, opt := range opts {
+		opt.ApplyHost(&c)
+	}
+	return c
+}
+
+// Start initializes reporting of host metrics using the supplied Config.
+func Start(c Config) error {
+	if c.MeterProvider == nil {
+		c.MeterProvider = global.MeterProvider()
+	}
+	h := &host{
+		meter: c.MeterProvider.Meter(
+			"host",
```

Comment:
Instrumentation needs to have the fully-qualified package name.

```suggestion
			"go.opentelemetry.io/contrib/instrumentation/host",
```